src/controllers/expenses.controller.js

```
@@ -0,0 +1,125 @@
+const { expensesService } = require('../services/expenses.service');
+const { usersService } = require('../services/users.service');
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const expenses = await expensesService.getExpenses();
```

getExpenses reads query params but doesn't validate them. Per the checklist you must return 400 when required/expected parameters are malformed. At minimum, if `userId` is provided and is not a numeric value you should return 400 with a JSON message. Also validate `from` and `to` are valid dates and return 400 if they are invalid before filtering. This prevents silent failures and ensures tests expecting 400 for bad queries pass.